pysnmp/proto/secmod/rfc3414/service.py

```diff
@@ -333,35 +333,40 @@ def __generateRequestOrResponseMsg(self, snmpEngine,
             0, scopedPDU, verifyConstraints=False, matchTags=False, matchConstraints=False
         )
 
-        # 3.1.6a
-        if securityStateReference is None and securityLevel in (2, 3):
-            if securityEngineID in self.__timeline:
-                (snmpEngineBoots, snmpEngineTime, latestReceivedEngineTime,
+        snmpEngineBoots = snmpEngineTime = 0
+
+        if securityLevel in (2, 3):
+            pdu = scopedPDU.getComponentByPosition(2).getComponent()
+
+            # 3.1.6.b
+            if pdu.tagSet in rfc3411.unconfirmedClassPDUs:
+                (snmpEngineBoots,
+                 snmpEngineTime) = mibBuilder.importSymbols('__SNMP-FRAMEWORK-MIB', 'snmpEngineBoots', 'snmpEngineTime')
```

Comment:
mibBuilder.importSymbols doesn't work in my program, and after I replace it to snmpEngine.msgAndPduDsp.mibInstrumController.mibBuilder.importSymbols it works. I'm not familiar with this project and can't find the reason so I can just tell you my test result.

Reply:
So far I can't see the difference because earlier in the code `mibBuilder` is initialized from `snmpEngine.msgAndPduDsp.mibInstrumController.mibBuilder`.

How exactly it does not work for you?